src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManagerTest.java

```diff
@@ -92,4 +95,120 @@ public void getNextQuestionsForFilter_appliesExclusionFilterForDeprecatedQuestio
     assertEquals(deprecatedFilter.getOperator(), Constants.BooleanOperator.NOT);
     assertEquals(deprecatedFilter.getValues(), Collections.singletonList("true"));
   }
+
+  @Test
+  public void generateRandomQuestions_returnsCorrectNumberOfQuestions() throws ContentManagerException {
+
+    // Arrange
+    int limit = 5;
+
+    var gameManager = new GameManager(
+        this.dummyContentManager,
+        this.dummyGameboardPersistenceManager,
+        this.dummyMapper,
+        this.dummyQuestionManager,
+        "latest"
+    );
+
+    List<IsaacQuestionPageDTO> questions = new ArrayList<>();
+    for (int i = 0; i < limit; i++) {
+      IsaacQuestionPageDTO question = createMock(IsaacQuestionPageDTO.class);
+      question.setSupersededBy(null);
+      questions.add(question);
+    }
+    var resultsWrapper = new ResultsWrapper(questions, (long) limit);
```

Comment:
`ResultsWrapper` is being used raw multiple times in this file, you need to specify the right type between `<>`. E.g. `ResultsWrapper<MyClass>`.